tests/unit/models/test_settings.py

```diff
@@ -0,0 +1,51 @@
+import os
+import tempfile
+from pathlib import Path
+
+from pytest import fixture
+
+from runner_manager.models.settings import (
+    ConfigFile,
+    Settings,
+    yaml_config_settings_source,
+)
+
+
+@fixture
+def test_settings_default_values():
+    settings = Settings()
+    assert settings.name == "runner-manager"
+    assert settings.redis_om_url == os.getenv("REDIS_OM_URL")
+    assert settings.github_base_url == os.getenv("GITHUB_BASE_URL")
+
+
+@fixture
+def temp_yaml_file():
+    yaml_data = """
+    name: test-runner-manager
+    redis_om_url: redis://localhost:6379/0
+    github_base_url: https://github.com
+    """
+    yaml_file = tempfile.NamedTemporaryFile(mode="w", delete=False)
+    yaml_file.write(yaml_data)
+    return yaml_file.name
+
+
+def test_yaml_config(temp_yaml_file):
+    config_file = ConfigFile(config_file=Path(temp_yaml_file))
+    settings = Settings()
+    settings.__config__.config = config_file
+    config_data = yaml_config_settings_source(settings)
+    assert config_data["name"] == "test-runner-manager"
+    assert config_data["redis_om_url"] == "redis://localhost:6379/0"
+    assert config_data["github_base_url"] == "https://github.com"
```

Comment:
This is the expected behavior from which we will be working with when using settings:
```suggestion
    os.environ['CONFIG_FILE'] = temp_yaml_file
    # settings should automatically look for the config file
    # environment variable and load the file if it exists
    settings = Settings()
    yaml_file = yaml.load(temp_yaml_file)
    assert settings["name"] ==  yaml_file["name"]
    assert settings["redis_om_url"] ==  yaml_file["redis_om_url"]
    assert settings["github_base_url"] == yaml_file["github_base_url"]
    
```